libtenzir/src/tql2/eval_impl.cpp

```diff
@@ -152,6 +154,48 @@ auto evaluator::eval(const ast::root_field& x) -> series {
   return null();
 }
 
+auto evaluator::eval(const ast::index_expr& x) -> series {
+  auto value = eval(x.expr);
+  auto index = eval(x.index);
+  if (auto number = index.as<int64_type>()) {
```

Comment:
When using a negative number for the list index, the counting starts from the end, which is nice.
But going out of bounds starting from the end results in an assertion instead of a warning.

You can reproduce the assertion with this pipeline:
```
// experimental-tql2
source { }        
test = [2,null,4]
this = {
   t: test[-4]
 }

```

Reply:
Nice catch. There was indeed a missing check for negative indices that are out of bounds.